lesson3/README.md

```diff
@@ -0,0 +1,55 @@
+
+### Code
+
+Now that you know the concept behind making a `ListView` work, read through [this guide](https://github.com/codepath/android_guides/wiki/Using-an-ArrayAdapter-with-ListView) to learn how to code it (you can skip the section about row view recycling if you want).
+
+Here is some basic sample code using the built-in `ArrayAdapter` to get you started:
+```java
+// Make a friends ArrayList
+ArrayList<String> friends = new ArrayList<>();
+friends.add("David");
+friends.add("Bill");
+
+// Use Google's ArrayAdapter, pass in the simple_list_item_1 layout and our list of friends
+ArrayAdapter<String> itemsAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, friends);
+
+// Initialize our ListView, linking it to the friendsListView in this fragment's layout file
+ListView listView = (ListView) findViewById(R.id.friendsListView);
+// Link this ListView to our adapter
+listView.setAdapter(itemsAdapter);
+```
+
+## Debugging
+Android Studio has a built-in debugger that allows you to set breakpoints in your code and see the values of variables during runtime.  This is extremely useful when debugging your code to see exactly what is going on at every line.
+
+[This official Android guide](https://developer.android.com/studio/debug/index.html) covers all the functionality of the AS debugger.  We'll go into the basics here.
+
+To make breakpoints, just click to the left of any line in your code.  To run your app in debugging mode, click the little bug icon next to the normal "Run" button.  Select your phone, and your app will get installed and run as normal.  If you hit a line with a breakpoint while using your app, the program will freeze and the debug window will pop up.  At each line before your breakpoint, variables and their values will be displayed.  You can mouse over specific variables or boolean expressions to see their values.  Using the buttons on the top of the debug window, you can step over each line of code to see what happens step-by-step, or resume your program as normal.
+
+## Assignment
+Before next class, make a todo app.  You should have:
+- A way to add new todo items.
+- A display of all the current todo items.
+- A way to delete items when the user completes them.
+- A way to edit an item.
+
+Stretch goals (optional):
+- Use a custom `Adapter`! You could make an icon that changes when the item is completed, or make each item have an inline edit/complete button.

```

Comment:
I think this could be required. The assignment will take like 15 minutes otherwise.